File: domain-check.js
#!/usr/bin/env node
/**
 * @author Chris Olstrom
 */

console.log('DomainCheck v2011.12.04 -- Copyright (c) 2010-2011 Chris Olstrom');

var argv = require('optimist')
	.usage('Usage: $0')
	.default({
		'api'	:'reseller.enom.com/interface.asp'
	})
	.boolean('insecure')
	.alias('api','a')
	.alias('username','u')
	.alias('password','p')
	.demand('username')
	.demand('password')
	.describe('api','Which API to query')
	.describe('username','Username for target API')
	.describe('password','Password for target API')
	.describe('domain','Query status of this domain')
	.describe('insecure','Do not use SSL')
	.demand('domain')
	.argv;

File: domain-check.pl
#!/usr/bin/env perl

print 'DomainCheck v2010.07.09 -- Copyright (c) 2010 Chris Olstrom';

my %config;
$config{'protocol'}	= 'https';		# For testing, use 'http'.
$config{'server'}	= 'reseller.enom.com';	# For testing, use 'resellertest.enom.com'.
$config{'api'}		= 'interface.asp';
$config{'user'}		= ''; # your username
$config{'password'}	= ''; # your password

## Parse arguments, build and execute API call.
my $DomainList = join(',',@ARGV);
my @sink = `curl -s $config{'protocol'}://$config{'server'}/$config{'api'} -d \'UID=$config{'user'}\' -d \'PW=$config{'password'}\' -d 'Command=Check' -d 'DomainList=$DomainList'`;

## Scrape the returned data for the requested information.
my (@domain,@status);
foreach my $record (@sink) {
	if ( $record =~ m/Domain(\d+)=(.+)\n/m ) {
		$domain[$1] = $2;
	}
	if ( $record =~ m/RRPText(\d+)=(.+)\n/m ){
		$status[$1] = $2;
	}
}

## Format everything and dump it to screen.
print "\nID\tDomain\t\t\tStatus";
print "\n--\t------\t\t\t------";
for (my $id = 1; $id <= scalar(@ARGV); $id++) {
	print "\n$id\t$domain[$id]\t\t\t\t$status[$id]";
}
print "\nDone checking ".scalar(@ARGV)." domains.\n";
